Composer/packages/server/src/services/telemetry.ts

```diff
@@ -43,10 +57,16 @@ if (instrumentationKey) {
   AppInsights.defaultClient.addTelemetryProcessor((envelope: AppInsights.Contracts.Envelope): boolean => {
     const { sessionId, telemetry, composerVersion, userId, architecture, cpus, memory } = getTelemetryContext();
 
+    const data = envelope.data as AppInsights.Contracts.Data<AppInsights.Contracts.RequestData>;
+
+    if (alwaysTrackEvents.includes(data.baseData.name as TelemetryEventName) && !telemetry?.allowDataCollection) {
+      stripPII(data);
+      return true;
```

Comment:
I like the thought process here by returning true before the other properties get set; however, we do need some of them to be set like data.baseData.properties.toolName